contracts/sentinel/PCVSentinel.sol

```diff
@@ -0,0 +1,131 @@
+     */
+    function isGuard(address guard) external view override returns (bool) {
+        return guards.contains(guard);
+    }
+
+    /**
+     * @notice returns a list of all guards
+     */
+    function allGuards() external view override returns (address[] memory all) {
+        all = new address[](guards.length());
+
+        for (uint256 i = 0; i < guards.length(); i++) {
+            all[i] = guards.at(i);
+        }
+
+        return all;
+    }
+
+    // ---------- Governor-or-Admin-Or-Guardian-Only State-Changing API ----------
+
+    /**
+     * @notice adds a guard contract to the PCV Sentinel
+     * @param guard the guard-contract to add
+     */
+    function knight(address guard)
+        external
+        override
+        isGovernorOrGuardianOrAdmin
+    {
+        guards.add(guard);
+
+        // Inform the kingdom of this glorious news
+        emit GuardAdded(guard);
+    }
+
+    /**
+     * @notice removes a guard
+     * @param traitor the guard-contract to remove
+     */
+    function slay(address traitor)
+        external
+        override
+        isGovernorOrGuardianOrAdmin
+    {
+        guards.remove(traitor);
+
+        // Inform the kingdom of this sudden and inevitable betrayal
+        emit GuardRemoved(traitor);
+    }
+
+    // ---------- Public State-Changing API ----------
+
+    /**
+     * @notice checks the guard and runs its protec actions if needed
+     * @param guard the guard to activate
+     */
+    function protec(address guard) external override nonReentrant {
+        require(guards.contains(guard), "Guard does not exist.");
+
+        if (IGuard(guard).check()) {
+            (
+                address[] memory targets,
+                bytes[] memory calldatas,
+                uint256[] memory values
+            ) = IGuard(guard).getProtecActions();
+
+            for (uint256 i = 0; i < targets.length; i++) {
+                require(targets[i] != address(this), "Cannot target self.");
+                (bool success, bytes memory returndata) = targets[i].call{
+                    value: values[i]
+                }(calldatas[i]);
+                Address.verifyCallResult(
+                    success,
+                    returndata,
+                    "Guard sub-action failed with empty error message."
+                );
+            }
+
+            emit Protected(guard);
+        } else {
+            revert("No need to protec.");
+        }
+    }
+
+    /**
+     * @dev receive() and fallback() have been added and made payable for cases where the contract
+     * needs to be able to receive eth as part of an operation - such as receiving an incentivization
+     * (in eth) from a contract as part of operation. For similar (and not unlikely) edge cases,
+     * the contract also has the capability of sending eth inside when instructed by a guard to do so.
+     */
+    receive() external payable {}
+
+    fallback() external payable {}
```

Comment:
These two functions cannot exist as long as this contract is not a PCV deposit. If it's a PCV deposit then you can add them because you have a way to remove the eth that this contract could potentially receive. Right now, any eth that is received is locked forever.

Reply:
I think its fine

This contract isn't meant to hold meaningful amounts of ETH. Any "safe ETH" moves will be sent to DAO timelock

Reply:
It also can specify how much eth to send in the `protec()` function so it's good to go at this point.